src/main/java/com/bootcamp/demo/service/LoginService.java

```diff
@@ -0,0 +1,46 @@
+package com.bootcamp.demo.service;
+
+import com.bootcamp.demo.model.Session;
+import com.bootcamp.demo.model.User;
+import com.bootcamp.demo.repository.SessionRepository;
+import com.bootcamp.demo.repository.UserRepository;
+
+import java.time.Instant;
+
+public class LoginService implements ILoginService {
```

Comment:
Pentru a apela functia de LoginService după în Controller e mai util să facem o clasă singleton, iar pentru asta ar mai fi nevoie de o instanță.
Ne ajută link-ul ăsta: https://www.journaldev.com/1377/java-singleton-design-pattern-best-practices-examples

Reply:
Nu sunt sigura ca inteleg de ce am avea nevoie de Singleton... Eu stiu ca acest Design Pattern se foloseste cand o clasa se foloseste in mai multe locuri in aplicatie, si sa prevenim sa nu apara noi instante, adica sa o folosim mereu pe aceeasi. Dar noi clasa de ServiceLogin o folosim doar in Controller si atat.

Reply:
Ai dreptate, mă gândeam la principiile SOLID, dar nu are sens să complicăm în acest caz că o folosim doar într-un loc.